### PR TITLE
Fix.nested.suites

### PR DESCRIPTION
**Pull request by grawk**
Nested suites were failing due to faulty logic in the `suite end` event. Take into account total # of suites before killing driver. Add nested suites to tests. Add couple more scripts to package.json

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,8 +4,10 @@
   "description": "Wrapper to run nemo/mocha suites",
   "main": "index.js",
   "scripts": {
-    "test": "npm run lint && ./bin/nemo-runner -B test -G @suite1",
-    "test:parallel": "./bin/nemo-runner -B test -G @suite1,@suite2,@suite3,@suite4,@suitex -P chrome,firefox",
+    "test": "npm run lint && npm run nemo",
```
Comment (skratchdot) [inline package.json:7]: It'd be nice to have this run all the nemo scripts (debug and parallel)- or at least have travis run them.

File: package.json [hunk 2]
```diff
@@ -4,8 +4,10 @@
...
+    "nemo": "DEBUG=nemo-runner* ./bin/nemo-runner -B test -G @suite1",
+    "nemo:d": "./bin/nemo-runner -B test -G @suite1 --debug-brk",
```
Comment (skratchdot) [inline package.json:9]: I tried running `npm run nemo:d` and it failed due to the fact that I'm running node 8.  Not a big deal, just calling it out.

**Comment by skratchdot** [comment]
This looks good to me.  I'm not super familiar with the codebase (or edge cases of working w/ mocha like this), but the logic in the diff looks good.  I'll let someone else click merge (or I'll do it after a little time passes) :)

**Review by gabrielcsapo — APPROVED** [review]

**Review by skratchdot — APPROVED** [review]

File: package.json (hunk 2 above)
Comment (grawk) [reply package.json:9]: Oh right. Yeah the flags change in node8.. need to account for that in the code itself as well. Filing an issue for that. Thanks!

File: package.json (hunk 1 above)
Comment (grawk) [reply package.json:7]: True. Good enhancement.

**Pull request merged**